internal/gatewayapi/route.go

```diff
@@ -1261,9 +1261,11 @@ func (t *Translator) processDestination(backendRefContext BackendRefContext,
 			Protocol:    protocol,
 			Endpoints:   endpoints,
 			AddressType: addrType,
+			IPFamily:    getIPFamily(envoyProxy),
```

Comment:
`getIPFamily` gets the value from the `envoyProxy` which is the listener's IP Family, not the backend's. This value should be coming from https://kubernetes.io/docs/concepts/services-networking/dual-stack/#dual-stack-service-configuration-scenarios

Reply:
Thanks for the feedback! I’ve updated the code to use a new IPFamily field
cc @zirain